### PR TITLE
Add target-setting runs

### PR DESCRIPTION
**Pull request by runame**
This PR adds the current configurations of the target-setting runs, including Jax and PyTorch implementations of the AdamW, NAdamW, and Nesterov optimizers.

Still missing:
- Interface for label smoothing and dropout which is accessible to the user (currently affects the ImageNet-ResNet and the WMT target-setting run),
- target-setting runs for workloads which are not fully implemented in this repo (PyTorch Criteo1TB DLRM-Small, Jax FastMRI + all Librispeech workloads).

### CODE REVIEW

**Review by znado — APPROVED** [review]

File: target_setting_runs/jax_nesterov.py [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+                         rng: spec.RandomState) -> spec.OptimizerState:
+  """Creates a Nesterov optimizer and a learning rate schedule."""
+  del model_params
+  del model_state
+  del rng
+
+  # Create optimizer.
+  params_zeros_like = jax.tree_map(lambda s: jnp.zeros(s.shape_tuple),
+                                   workload.param_shapes)
+  opt_init_fn, opt_update_fn = static_inject_hyperparams(sgd)(
```
[the remark this answers is not in the document]
Comment (runame) [reply target_setting_runs/jax_nesterov.py]: Ah nice, I used it because as you say `if t < warmup_steps:` cannot be jitted, but just using `optax.join_schedules` is much better. I have also noticed that the optax implementation of the linear warmup (`optax.linear_schedule`) and the polynomial decay (`optax.polynomial_schedule`) seem to be equivalent to the init2winit implementation at first glance, so I'm using them now which requires less code. Or is there a difference I have missed? Not sure why they were re-implemented in init2winit.

File: target_setting_runs/jax_nesterov.py [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+
+  optimizer_state = {
+      'optimizer_state': jax_utils.replicate(optimizer_state),
+      'lr_schedule_fn': lr_schedule_fn
+  }
+
+  return optimizer_state, opt_update_fn
+
+
+def create_lr_schedule_fn(
```
[the remark this answers is not in the document]
Comment (runame) [reply target_setting_runs/jax_nesterov.py:39]: No strong preference, do you still think it makes sense, now that only `create_lr_schedule_fn` remains and the rest was removed?

File: target_setting_runs/jax_nesterov.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (znado) [reply target_setting_runs/jax_nesterov.py]: nope they should be the same, we kept our own because we both implemented them in parallel.

File: target_setting_runs/jax_nesterov.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (znado) [reply target_setting_runs/jax_nesterov.py:39]: nope LGTM
